### ISSUE TITLE
Experiment with Model size and other hyperparameters

### ISSUE THREAD

**Issue by tanzir5** [odissei-lifecourse/life-sequencing-dutch#20]
Currently we are using the default parameter values and the model size from L2V. A good idea will be to read the appendix of L2V to see what their experiments already yielded and decide for our use case how to set the optimal training parameters.

This is the final pretraining hparams used by L2V -- https://github.com/SocialComplexityLab/life2vec/blob/5601191a08a3feae97b4641776692916400c8ed7/conf/experiment/pretrain.yaml

This sheet contains their hparam search -- https://docs.google.com/spreadsheets/d/1url1ECO-bKWWxrIlyQk2Hpt7KoSFWpo4DK6l7p8gRAI/edit?gid=504672320#gid=504672320

**Comment by tanzir5** [comment]
Epochs -- we have trained our model for 30 epochs, using data of 1.8 million people. We plan to let the model train for as long as it can until early stopping hits. 

L2v  had 3.2 million people in their dataset . L2V trained for 30 epochs where each epoch had 30K random people. Therefore they only use 1 million people each seen once. Compared to this, our model sees 1.8 million people 30 times. Therefore there is nothing else to be done for epoch except for letting it run as long as it wants.

**Comment by tanzir5** [comment]
Batch_size --> L2V used 8. They had sequences of length 2500, we are dealing with 512 and much better gpus, our batch_size has been 256 for medium2x model iirc. 

- [ ] Verify batch_size on OSSC. 

The batch_size should be maximum possible our gpus can handle to make the training efficient. But learning rate should also be changed along with batch_size. More in the comment for learning_rate.

**Comment by f-hafner** [comment]
closing as outdated
